index/scorch/segment/zap/segment.go

```diff
@@ -118,6 +120,13 @@ func (s *Segment) SizeInBytes() uint64 {
 	sizeInBytes += len(s.fieldsOffsets) * 8 /* size of uint64 */
 	sizeInBytes += 8                        /* size of refs -> int64 */
 
+	for _, entry := range s.fieldDvIterMap {
+		sizeInBytes += int(unsafe.Sizeof(entry)) + 2 /* size of uint16 */
```

Comment:
I believe since the values in the fidldDvIterMap are pointers, then unsae.Sizeof(entry) gives you the size of the pointer and not the struct itself.  Is that what you intended?

Reply:
yes, and then upon valid pointers compute the size of the referenced.

Reply:
in that case, i'd prefer we just assume 8 bytes for a pointer, rather than using unsafe